### PR TITLE
chore: Improve changelog with breaking changes

### LINKED ISSUE #1674: Add breaking changes to the release note generator
While we haven't committed to a stable API, it is useful for developers to understand what breaking changes have been made for releases. This requires a change in the change-log generation.

### PR DESCRIPTION
**Pull request by marlass**
With this changes in generated changelog we will have informations about breaking changes below new features table. Should help developers with migrating from older spartacus versions.

Closes #1674

### CODE REVIEW

File: scripts/templates/breaking-change.ejs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<li>
+<%= content %>
+<%
+      for (const reference of commit.references) {
```
Comment (Pucek9) [inline scripts/templates/breaking-change.ejs]: ejs supports js build in functions like forEach so we can use it:

```
commit.references
  .filter(reference => !reference.action || !reference.issue)
  .forEach(reference => {
    // ...
  });
```
PS. I don't know are arrow functions allowed here

File: scripts/changelog.ts [hunk 2]
```diff
@@ -85,7 +86,17 @@ export default function run(args: ChangelogOptions, logger: logging.Logger) {
             if (tags && tags.find(x => x == args.to)) {
               toSha = chunk.hash as string;
             }
-
+            const notes: any = chunk.notes;
+            if (notes.length) {
+              notes.forEach(note => {
+                if (note.title === 'BREAKING CHANGE') {
```
Comment (Pucek9) [inline scripts/changelog.ts]: you can use filter here

File: scripts/changelog.ts [hunk 3]
```diff
@@ -85,7 +86,17 @@ export default function run(args: ChangelogOptions, logger: logging.Logger) {
             if (tags && tags.find(x => x == args.to)) {
               toSha = chunk.hash as string;
             }
-
+            const notes: any = chunk.notes;
+            if (notes.length) {
```
Comment (Pucek9) [inline scripts/changelog.ts]: Array.isArray(notes)

File: scripts/templates/breaking-change.ejs (hunk 1 above)
Comment (marlass) [reply scripts/templates/breaking-change.ejs]: Tried and this doesn't work. Using regular function also ended up with template compilation error.

**Pull request merged**